### ISSUE TITLE
How to maintain the FIFOQueue when training

### ISSUE THREAD

**Issue by UpCoder** [tensorflow/tensorflow#30250]
Hi, I want to use FIFOQueue to keep K elements in training process.
```python
    input_tensor = tf.placeholder(tf.float32, shape=[None, 224, 224, 3], name='x')
    model = SEVGGModel(input_tensor, input_tensor)
    print(model.output_feature, model.output_feature_map, model.output_feature_norm)
    import os
    os.environ["CUDA_VISIBLE_DEVICES"] = "3"
    sess = tf.Session()
    sess.run(tf.initialize_all_variables())
    # coord = tf.train.Coordinator()
    # threads = tf.train.start_queue_runners(coord=coord, sess=sess)

    print(sess.run(model.last_ele_queue.size()))
    print('ref is ', model.last_ele_queue.queue_ref)
    with tf.control_dependencies([model.enqueue_op]):
        # output_tensor = tf.cond(
        #     tf.equal(model.last_ele_queue.size(), 0),
        #     lambda: 0.,
        #     lambda: model.last_ele_queue.dequeue_many(30)
        # )
        loss_tensor = model.build_loss(True)
    for i in range(100):
        feed_dict = {
            input_tensor: np.random.random([30, 224, 224, 3]) * i
        }
        # sess.run(model.enqueue_op, feed_dict=feed_dict)  ## with or without
        print(i, sess.run(model.last_ele_queue.size()))
        _, loss_value = sess.run([model.enqueue_op, loss_tensor], feed_dict=feed_dict)
        # print(np.shape(values), np.max(values), np.min(values))
        print('loss_value ', loss_value)
    print(sess.run(model.last_ele_queue.size()))
```
the loss value can be see as the sum of all elements in Queue and the feed_dict is the element in the Queue..
I found when I use the code sess.run(model.enqueue_op, feed_dict=feed_dict), I think the loss value should be double. But the result is not. The result of with or without this line are same.

So can you help me explain it?

Thanks you!

**Comment by gadagashwini** [comment]
@UpCoder Please provide details about what platform you are using (operating system, architecture). Also include your TensorFlow version. Also, did you compile from source or install a binary?

In order to expedite the trouble-shooting process, please provide a full code snippet to reproduce the issue reported here. If you are unclear what to include see the issue template displayed in [the Github new issue template](https://github.com/tensorflow/tensorflow/issues/new/choose).

We ask for this in the issue submission template, because it is really difficult to help without that information. Thanks!

**Comment by gadagashwini** [comment]
Automatically closing due to lack of recent activity. Please update the issue when new information becomes available, and we will reopen the issue. Thanks!
